### ISSUE TITLE
"no such index" when trying to upload a sample metadata file

### ISSUE THREAD

**Issue by slead** [Esri/geoportal-server-catalog#1]
Following the instructions at https://github.com/Esri/geoportal-server-catalog/wiki/Installing-Geoportal-Server-2.0.0 I've installed ElasticSearch, which is working at http://localhost:9200/ so now I'm trying to "Upload a few metadata records".

This returns the error message "no such index":

![screen shot 2016-05-18 at 3 35 26 pm](https://cloud.githubusercontent.com/assets/133596/15348434/3a257a18-1d0e-11e6-9ee1-b4ecf50066ac.png)

The [geoportal.log.txt](https://github.com/Esri/geoportal-server-catalog/files/269708/geoportal.log.txt) is available here if that helps. Thanks

**Comment by umacgillivray** [comment]
Hi Steven. 
Here is the original error:
NoNodeAvailableException[None of the configured nodes are available: [{#transport#-1}{127.0.0.1}{localhost/127.0.0.1:9300}]] at org.elasticsearch.client.transport.TransportClientNodesService.ensureNodesAreAvailable(TransportClientNodesService.java:290)

**Comment by umacgillivray** [comment]
A couple of questions:
Is Geoportal on the same machine as Elasticsearch?
Are you on Windows or Linux or Mac ?

http://localhost:9200
will provide some info on Elasticsearch
Is the cluster_name "elasticsearch" ?
Is the version number "2.3.1" ?

thx

**Comment by slead** [comment]
Hi Urban,

I'm on Windows 7, with Elasticsearch running on port 9200. It looks like I installed Elasticsearch 2.3.2 - is that likely to be a problem?

http://localhost:9200 returns:

```
{
  "name" : "Guido Carosella",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "2.3.2",
    "build_hash" : "b9e4a6acad4008027e4038f6abed7f7dba346f94",
    "build_timestamp" : "2016-04-21T16:03:47Z",
    "build_snapshot" : false,
    "lucene_version" : "5.5.0"
  },
  "tagline" : "You Know, for Search"
}
```

(http://localhost:9300 returns "This is not a HTTP port" for what it's worth.)

Thanks for any tips.

**Comment by slead** [comment]
Hmm, I just tried again using the same file and it worked this time. I'll keep an eye on it and will report back if the problem reoccurs. Thanks!

**Comment by umacgillivray** [comment]
When Tomcat starts, Geoportal checks Elasticsearch: if the "metadata" index doesn't exist it gets created. Maybe Elasticsearch wasn't running when Tomcat was initially started, so the "metadata" index didn't get created, a later restart of Tomcat resolved the issue. We can probably modify our code mitigate this issue in the future.

**Comment by slead** [comment]
I just tried again with ES stopped, and got a slightly different error when trying to upload an XML:

> None of the configured nodes are available: [{#transport#-1}{127.0.0.1}{localhost/127.0.0.1:9300}]Upload 

There was no error message when trying to *search* for metadata, so maybe the geoportal code should check for a live ES service on startup, then show an error message at that point.

**Comment by smrgeoinfo** [comment]
I'm seeing same behavior, using a 2.5.1 SNAPSHOT build (CINERGI fork). netstat says its listening on 9300 (tcp6--does that matter?), and the index exists (curl http://localhost:9200/metadata returns the mapping file.
